### DISCUSSION TITLE
Adding Games to Sunshine

### DISCUSSION THREAD

**Discussion by Silas888** [LizardByte/community#136] [Sunshine]
I'm having an issue when attempting to launch games within Moonlight from Sunshine. The majority of the time I'll get the following error: "Connection Error Server Error." I'll take for example God of War. I have it configured in Sunshine the same way I have it in GeForce Experience. In the command section of Sunshine I have listed: "D:\Games\God of War\GoW.exe". If I move the same line to Detached Commands the desktop will show but the game won't be launched. Funny thing is that I also have Playnight configured and all games work when I launch them through Playnight . I'm thinking that this must then be related to how Sunshine launches the exe. Any help would be appreciated.

**Comment by Silas888** [comment]
I should incvlude the log file. The same log entry occurs for every game that fails to launch
[2023:01:11:17:27:14]: Info: Executing: ["D:\Games\God of War\GoW.exe"] in ["D:\Games\God of War"]
[2023:01:11:17:27:14]: Error: Failed to launch process: 740
[2023:01:11:17:27:14]: Warning: Couldn't run ["D:\Games\God of War\GoW.exe"]: System: Invalid argument

**Reply by nordeep** [reply]
Seems like quotes are not needed in the game executable path.
Should be D:\Games\God of War\GoW.exe
In the log in looks like:
```
[2023:01:26:12:43:46]: Info: Executing: [C:\Games\Disco Elysium\Disco Elysium.exe] in ["C:\Games\Disco Elysium"]
[2023:01:26:12:43:46]: Info: C:\Games\Disco Elysium\Disco Elysium.exe running with PID 7964
```

**Reply by Silas888** [reply]
Even if I remove the quotes, I get the same result. It's different for different games. Some of them work some of them don't. Others require a detached command. The one thing they all have in common is it every single one worked in GeForce experience.

**Reply by gschintgen** [reply]
> Even if I remove the quotes, I get the same result. It's different for different games.

What are your **exact** entries in the Web UI?

"Detached Commands": Empty or not? Full path or not? With or without quotes?
"Command": Empty or not? Full path or not? With or without quotes?
"Working Directory": Empty or not? Full path or not? With or without quotes?

Anyway I'm new to Sunshine myself and had a very similar issue. I had done the following (in Windows): shift+right-click the executable, copy as path, paste into "Detached Commands". Same procedure for the working directory. This results in full paths for both text fields, both entries with quotes. That did *not* work. The solution in my case was quite simple: I just removed the manual entry of the working directory and left the detached commands as they were (full path, with quotes, e.g. `"C:\Program Files ..... Cool Game.exe"`).

**Reply by Silas888** [reply]
No detached command 
Command: D:\Games\God of War\GoW.exe
Working directory: D:\Games\God of War

I was finally able to get it to work by creating a batch file, sticking it in the game folder where the executable is, and reference the batch file as a command in Sunshine. I don't feel like I should've had to have done that though, as a batch file was not necessary with GeForce Experience.

**Reply by buthig666** [reply]
Same here.  I get kicked to desktop no matter what app I add.

**Reply by Silas888** [reply]
Seems to be a different experience for everyone.

**Reply by gschintgen** [reply]
> Same here. I get kicked to desktop no matter what app I add.

This won't lead anywhere if you don't provide any details. (See my previous comment.)

I'd like to add that my comments are based on v0.18.

I recommend to first try as detached command (see documentation) with an unspecified working directory  In a second stage you could then try if it also works as a regular command.

**Comment by ReenigneArcher** [comment]
Try adding `cmd /c` in front of your command, i.e. `cmd /c "GoW.exe"`

It still may need to be detached.

**Comment by shackrock** [comment]
For what it's worth, it's the same for me. I will try the cmd /c next - because nothing else has ever worked for any game for me.  I just use a wireless keyboard to open the game after the desktop shows up, best I can do for now until I find something effective at least!

[a comment by an automated account is not shown]
